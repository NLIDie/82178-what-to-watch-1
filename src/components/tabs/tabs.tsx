import * as React from "react";

// Components
import {TabsItem} from "_Components/tabs-item/tabs-item";

// Hooks
import {useActiveItem} from "_Hooks/use-active-item";

// Types
import {TProps} from "./tabs.types";

export function Tabs(props: TProps): JSX.Element {
  const {
    className = `movie-card__nav`,
    tabs = [`Overview`, `Details`, `Reviews`],
    children
  } = props;

  const [activeTab, handleTabClick] = useActiveItem<string>(tabs);

  const tabList: JSX.Element[] = tabs.map<JSX.Element>((tab: string) => (
    <TabsItem
      key={tab}
      tab={tab}
      onClick={handleTabClick}
      isActive={tab === activeTab}
    />
  ));

  return (
    <div>
      <nav className={`movie-nav ${className}`}>
        <ul className="movie-nav__list">{tabList}</ul>
      </nav>

      {children(activeTab)}
    </div>
  );
}
