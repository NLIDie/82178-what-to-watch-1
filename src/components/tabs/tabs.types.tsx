export type TProps = {
  className?: string;
  tabs: string[];
  children: (activeTab: string) => JSX.Element;
};
