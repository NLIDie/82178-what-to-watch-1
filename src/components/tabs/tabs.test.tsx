import * as React from "react";
import {create} from "react-test-renderer";

// Components
import {Tabs} from "./tabs";

describe(`Components/Tabs`, () => {
  it(`Correct renderer the component`, () => {
    const tree = create(
      <Tabs tabs={['tab1', 'tab2']}>
        {(tab) => <div>tab</div>}
      </Tabs>
    );

    expect(tree).toMatchSnapshot();
  });
});
