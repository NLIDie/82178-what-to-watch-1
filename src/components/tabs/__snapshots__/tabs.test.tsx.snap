// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Tabs Correct renderer the component 1`] = `
<div>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          tab1
        </a>
      </li>
      <li
        className="movie-nav__item "
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          tab2
        </a>
      </li>
    </ul>
  </nav>
  <div>
    tab
  </div>
</div>
`;
